### ISSUE TITLE
Unable to launch MicrosoftD365 login page using cypress. Url get into loop and gives 302 error code

### ISSUE THREAD

**Issue by umaalagesan** [cypress-io/cypress#26869]
### Current behavior

Steps:

Just trying to launch the mD365 application through cypress tool.
<img width="1728" alt="Screenshot 2023-05-28 at 17 13 31" src="https://github.com/cypress-io/cypress/assets/106514702/32426406-a683-4143-8a92-555324163df4">

Also tried **experimentalModifyObstructiveThirdPartyCode** flag as true and false in config file

### Desired behavior

https://github.com/cypress-io/cypress/assets/106514702/b2bfdb29-0ef4-401c-ad31-1d04a16434a2

### Test code to reproduce
```js
describe('Authenticate', () => {

  it('logs into login.microsoftonline through a user web app', () => {
    cy.visit('https://ewa-qad2f13c7892891sssss884devaos.axcloud.dynamics.com/');
    cy.get('button#sign-in-dropdown').click();
    cy.get('button#sign-in-redirect').click();

    cy.origin('login.microsoftonline.com', () => {
      cy.get('input[type="email"]').type(Cypress.env('aad_username'));
      cy.get('input[type="submit"]').click();
    });

    // depending on the user and how they are registered with microsoft, the origin might go to live.com
    cy.origin('login.live.com', () => {
      cy.get('input[type="password"]').type(Cypress.env('aad_password'), {
        log: false
      });
      cy.get('input[type="submit"]').click();
      cy.get('#idBtn_Back').click();
    });

    cy.get('#welcome-banner').should(
      'contain',
      `Welcome ${Cypress.env('aad_name')}`
    );
  });
})
```
### Cypress Version

10.10.0,12.0.0 etc 

### Node version

v20.1.0

### Operating System

12.0.1 (21A559)

### Debug Logs

```shell
@AtofStryker
Has shared the code in one of the prior issues. But I missed the issue id , also that issue was closed, So oepning new issue as i tried in latest cypress version also
```

**Comment by umaalagesan** [comment]
#26869 is related to https://github.com/cypress-io/cypress/issues/1342#issuecomment-1339917018

**Comment by AtofStryker** [comment]
Hi @umaalagesan . Thank you for opening an issue. Do you know if this works when running 'electron' or if your browser has any corporate security policies implemented? I also created a r[eproduction repo](https://github.com/AtofStryker/cypress-issue-26869), but it looks like the instance is down? Would you be able to fill out some of the details to help us reproduce the problem?

**Comment by umaalagesan** [comment]
Hi @AtofStryker  It happens even for normal personal microsoft login. These are the differences I found for normal chrome and  cypress initiated chrome instance. Is it related to certificate issue ? Kindly help with this, would need to finalize the tool. Thank you.


<img width="1728" alt="Screenshot 2023-06-01 at 12 01 58" src="https://github.com/cypress-io/cypress/assets/106514702/fad7d551-66cb-4b0b-8c41-8cacb24c8b9d">
<img width="1728" alt="Screenshot 2023-06-01 at 12 02 02" src="https://github.com/cypress-io/cypress/assets/106514702/d6afda5b-1d29-4beb-a7cd-024c4fb97325">
![Uploading Screenshot 2023-06-01 at 12.02.12.png…]

<img width="1728" alt="Screenshot 2023-06-01 at 12 01 44" src="https://github.com/cypress-io/cypress/assets/106514702/b947d744-7c3d-4290-8dd8-fb0b086e2a80">

[Archive.zip](https://github.com/cypress-io/cypress/files/11622205/Archive.zip)

**Comment by umaalagesan** [comment]
CC  :radha.maheshwari@eyewa.com

**Comment by AtofStryker** [comment]
@umaalagesan it is unlikely related to the certificate as cypress needs to self sign certs to set your application URL as top inside the test runner. Just to reiterate some of the diagnostic steps from above:
* Do you know if this works when running 'electron' or if your browser has any corporate security policies implemented?
*  Are you able to help fill out the missing details in the [reproduction repo](https://github.com/AtofStryker/cypress-issue-26869)? The instance posted in the description is down and I am unable to access it or run the test due to missing credentials.

We'd love to help unblock you and help with this issue, but we need the following above bullet points addressed so we can have the best chance at understanding the problem and providing a solution

**Comment by umaalagesan** [comment]
I have attached the zip folder script, it has working urls.
[Archive.zip](https://github.com/cypress-io/cypress/files/11627891/Archive.zip)

**Comment by umaalagesan** [comment]
Web page is not getting loaded. That is the main problem. Could you pls help me.

**Comment by AtofStryker** [comment]
@umaalagesan what I can it looks like you are getting framebusted. I am going to push your [Archive.zip](https://github.com/AtofStryker/cypress-issue-26869-2/) file to a reproduction repository and mark this issue as triaged. It will go into the team backlog to be weighed against other issues. Is this also what you are experiencing?

https://github.com/cypress-io/cypress/assets/3980464/ba24a578-09fb-4e33-8b1d-af0dea7f370a

**Comment by umaalagesan** [comment]
Yes, after clicking login, the page doesn't load. even manually tried in the chrome instance opened by cypress. Its not loading. 
<img width="1728" alt="Screenshot 2023-06-02 at 11 08 00" src="https://github.com/cypress-io/cypress/assets/106514702/e24f74b1-66d6-4d77-99db-6622759b4186">

**Comment by AtofStryker** [comment]
> Yes, after clicking login, the page doesn't load. even manually tried in the chrome instance opened by cypress. Its not loading. <img alt="Screenshot 2023-06-02 at 11 08 00" width="1728" src="https://user-images.githubusercontent.com/106514702/242795004-e24f74b1-66d6-4d77-99db-6622759b4186.png">

that almost looks like `experimentalModifyObstructiveThirdPartyCode` is off, but the [SRI](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) attributes might not be getting stripped here as this tab exists outside the test application. I am going to mark the issue as a framebusting problem.

**Comment by mschile** [comment]
Related GitHub discussion: https://github.com/cypress-io/cypress/discussions/25679
